### DISCUSSION TITLE
The Meter widget has been completely rebuilt for ttkbootstrap 1.0

### DISCUSSION THREAD

**Discussion by israel-dryer** [israel-dryer/ttkbootstrap#70] [Announcements]
The **Meter** widget has been completely rebuilt. This is related to issue #61 

See the new class file [here](https://github.com/israel-dryer/ttkbootstrap/blob/master/src/ttkbootstrap/widgets/meter.py)

**Scroll down** to see the code that produced these beautiful meters.

<img src="https://user-images.githubusercontent.com/42584742/141709579-5092d62f-bcd3-4857-9377-d9ed078ded48.png" width=50%>

```python
import tkinter as tk
import ttkbootstrap as ttk

root = tk.Tk()
style = ttk.Style("superhero")

ttk.Meter(
    master=root,
    metersize=180,
    padding=20,
    amountused=25,
    metertype='semi',
    subtext='miles per hour',
    interactive=True
).grid(row=0, column=0)

ttk.Meter(
    metersize=180,
    padding=20,
    amountused=1800,
    amounttotal=2600,
    subtext='storage used',
    textright='gb',
    bootstyle='info',
    stripethickness=10,
    interactive=True
).grid(row=0, column=1)

ttk.Meter(
    metersize=180,
    padding=20,
    stripethickness=2,
    amountused=40,
    subtext='project capacity',
    textright='%',
    bootstyle='success',
    interactive=True
).grid(row=1, column=0)

ttk.Meter(
    metersize=180,
    padding=20,
    amounttotal=280,
    arcrange=180,
    arcoffset=-180,
    amountused=75,
    textright='°',
    subtext='heat temperature',
    wedgesize=5,
    bootstyle='danger',
    interactive=True
).grid(row=1, column=1)

root.mainloop()
```

**Comment by jairuz** [comment]
How do I auto-update the value of 'amountused' coming from a serial data?
Is this possible?

**Reply by mihneati** [reply]
Did you succeed to find a solution? 
I try to pass a thread method result to amountused and it gives me error, saying in fact that you can only pass a float value

**Comment by Believer23** [comment]
i need a meter it wall auto update the value plese creat a new function like Auto updata 
and i wsa creating a dashboard app for my project so i need the option
